File: C/Gestion de Disques/6.3b.c
#include <stdio.h>
#include <stdlib.h>

int main() {
    FILE *f = fopen("tab.bin", "rb");
    if(!f) { perror("fopen"); return 1; }

    int *tab = NULL;
    int capacity = 10;
    int n = 0;
    tab = malloc(capacity * sizeof(int));

    int x;
    while(fread(&x, sizeof(int), 1, f) == 1) {
        if(n >= capacity) {
            capacity *= 2;
            tab = realloc(tab, capacity * sizeof(int));
        }
        tab[n++] = x;
    }

    printf("Tableau chargé (%d entiers) :\n", n);
    for(int i=0;i<n;i++) printf("%d ", tab[i]);
    printf("\n");

    free(tab);
    fclose(f);
    return 0;
}

File: Python/Gestion de Disques/6.3b.py
import struct
import os

tab = []

with open("tab.bin", "rb") as f:
    while True:
        bytes_read = f.read(4)  # lire 4 octets (int)
        if not bytes_read:
            break
        x = struct.unpack('i', bytes_read)[0]
        tab.append(x)

print(f"Tableau chargé ({len(tab)} entiers) : {tab}")